### PR TITLE
XW-1602 | Return data for international footer

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/XW-1602

### CODE REVIEW

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 1]
```diff
@@ -1,377 +1,610 @@
...
-						'key' => 'global-footer-wikia-company-overview-link-press'
+				'links' => [
+					[
+						'type' => 'link-image',
+						'image' => 'company-store-appstore',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-footer-wikia-community-apps-link-app-store'
+						],
+						'href' => 'https://itunes.apple.com/pl/developer/wikia-inc./id422467077'
```
Comment (rogatty) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Is `/pl/` for polish?
Yes, it is.

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 2]
```diff
@@ -1,377 +1,610 @@
...
 				],
-				[
-					'type' => 'link-text',
+			],
+		];
+
+		$data['fandom_overview'] = $this->getFandomOverview();
+		$data['follow_us'] = $this->getFollowUs();
+
+		if ( $this->lang == self::DEFAULT_LANG ) {
```
Comment (rogatty) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Why not `===`?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 1 above)
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Great question, I just moved it and didn't notice it. Probably it's not the best link for all languages...

**Comment by rogatty** [comment]
Two things and will LGTM

**Comment by kvas-damian** [comment]
@rogatty fixed, can you take 2nd look?

**Comment by rogatty** [comment]
:+1:

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 3]
```diff
@@ -385,4 +618,10 @@ private function getLicenseUrl() {
 
 		return $licenseUrl;
 	}
+
+	private function getHref( $hrefKey ) {
+		return isset( $this->hrefs[$this->lang][$hrefKey] )
```
Comment (Grunny) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: You can use the null coalescing operator here now we're on PHP 7, i.e.

``` php
return $this->hrefs[$this->lang][$hrefKey] ?? $this->hrefs['default'][$hrefKey];
```

**Comment by Grunny** [comment]
LGTM.

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 3 above)
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Doesn't it trigger a notice when one of the keys don't exist?

**Comment by Warkot** [comment]
LGTM. 👍

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 3 above)
Comment (Grunny) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: It shouldn't. `$foo['bar'] ?? $foo['foo'];` is equivalent to `isset( $foo['bar'] ) ? $foo['bar'] : $foo['foo'];`, it's just syntactic sugar.

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 4]
```diff
@@ -197,6 +197,10 @@ public function __construct( $wikiId, $lang = self::DEFAULT_LANG ) {
 		$this->lang = $lang;
 	}
 
+	public function setHrefs( $hrefs ) {
```
Comment (tomasznapieralski) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:200]: can this method be private?

File: includes/wikia/models/tests/DesignSystemGlobalFooterModelTest.php [hunk 5]
```diff
@@ -74,4 +74,180 @@ public function setLicensingAndVerticalDataProvider() {
...
+
+	public function testInternationalHeader() {
+		$footerModel = new DesignSystemGlobalFooterModel( 1234, 'en' );
+		$result = $footerModel->getData();
+
+		$this->assertNotEmpty( $result['wikia'] );
+		$this->assertNotEmpty( $result['fandom'] );
+		$this->assertArrayNotHasKey( 'international_header', $result );
+
+
```
Comment (tomasznapieralski) [inline includes/wikia/models/tests/DesignSystemGlobalFooterModelTest.php]: can you remove that extra line?

File: includes/wikia/models/tests/DesignSystemGlobalFooterModelTest.php [hunk 6]
```diff
@@ -74,4 +74,180 @@ public function setLicensingAndVerticalDataProvider() {
...
+		$footerModel = new DesignSystemGlobalFooterModel( 1234, 'en' );
+		$result = $footerModel->getData();
+
+		$this->assertCount( 4, $result['fandom_overview']['links'] );
+
+		$footerModel = new DesignSystemGlobalFooterModel( 1234, 'pl' );
+		$result = $footerModel->getData();
+
+		$this->assertCount( 1, $result['fandom_overview']['links'] );
+
```
Comment (tomasznapieralski) [inline includes/wikia/models/tests/DesignSystemGlobalFooterModelTest.php]: remove that extra line please

**Comment by tomasznapieralski** [comment]
left few comments, LGTM

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 4 above)
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:200]: Not really, it's used outside of this class. We could trick it with Reflection although I don't like this approach...

**Comment by hakubo** [comment]
LGTM

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php (hunk 4 above)
Comment (tomasznapieralski) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:200]: 👍

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**